src/session_handle.cpp

```diff
@@ -760,12 +760,14 @@ namespace {
 
 	unsigned short session_handle::listen_port() const
 	{
-		return sync_call_ret<unsigned short>(&session_impl::listen_port);
+		return sync_call_ret<unsigned short, unsigned short(session_impl::*)() const>
```

Comment:
This function is calling the overload that doesn't specify an interface. The public API right now doesn't expose any sort of interface handle so there's no way to have a public function which specifies one.